src/main/java/org/opensearch/ad/transport/PMMLResultRequest.java

```diff
@@ -0,0 +1,148 @@
+ */
+
+package org.opensearch.ad.transport;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+import java.io.IOException;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.ad.constant.CommonErrorMessages;
+import org.opensearch.ad.constant.CommonName;
+import org.opensearch.common.Strings;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+public class PMMLResultRequest extends ActionRequest implements ToXContentObject {
+    private String adID;
+    private String mlModelID;
+    private String[] featureNames;
+    private double[] features;
+
+    // Messages used for validation error
+    public static final String INVALID_FEATURE_NAME_MSG = "feature name vector is empty (in pmml result request)";
+    public static final String INVALID_FEATURE_MSG = "feature vector is empty (in pmml result request)";
+    public static final String INVALID_LENGTH_MSG = "feature name vector and feature vector have different lengths";
+
+    public PMMLResultRequest(String adID, String mlModelID, String[] featureNames, double[] features) {
+        super();
+        this.adID = adID;
+        this.mlModelID = mlModelID;
+        this.featureNames = featureNames;
+        this.features = features;
+    }
+
+    public PMMLResultRequest(StreamInput in) throws IOException {
+        super(in);
+        adID = in.readString();
+        mlModelID = in.readString();
+        int size1 = in.readVInt();
+        featureNames = new String[size1];
+        for (int i = 0; i < size1; i++) {
+            featureNames[i] = in.readString();
+        }
+        int size2 = in.readVInt();
+        features = new double[size2];
+        for (int i = 0; i < size2; i++) {
+            features[i] = in.readDouble();
+        }
+    }
+
+    public String getAdID() {
+        return adID;
+    }
+
+    public String getMlModelID() {
+        return mlModelID;
+    }
+
+    public String[] getFeatureNames() {
+        return featureNames;
+    }
+
+    public double[] getFeatures() {
+        return features;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(adID);
+        out.writeString(mlModelID);
+        out.writeVInt(featureNames.length);
+        for (String name : featureNames) {
+            out.writeString(name);
+        }
+        out.writeVInt(features.length);
+        for (double feature : features) {
+            out.writeDouble(feature);
+        }
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (adID == null || Strings.isEmpty(adID)) {
```

Comment:
Rather than setting a `validationException` var and returning it at the end, can we just return within each `if` block? That way it will fail faster without having to traverse the other conditions. Also, it will let you utilize previous conditions to do less checks (e.g., don't need the null checks on line 124).

Reply:
Could change to a cascading ifs too using `else if`

Reply:
I think this is just for testing use, the actual checks happen elsewhere, not in the request.

Reply:
Ok, just a minor thing for code readability & efficiency.